app/agents_v2/primary_agent/reasoning/reasoning_engine.py

```
@@ -58,6 +59,249 @@
+    ) -> None:
+        """
+        Performs unified comprehensive analysis combining query analysis, solution mapping,
+        and tool assessment into a single LLM call to reduce API usage.
+        """
+        with tracer.start_as_current_span("reasoning_engine.unified_analysis") as span:
+            try:
+                # Prepare unified analysis prompt
+                unified_prompt = f"""Analyze this customer support query comprehensively in one pass:
+
+**Query**: {query}
+**Context**: {json.dumps(context) if context else "No previous context"}
```

Ruff (0.12.2)

Undefined name `json`

(F821)